kwc-button.html

```diff
@@ -0,0 +1,378 @@
+        }
+        :host([transparent][type="primary"]) {
+            color: var(--color-kano-orange);
+        }
+        :host([transparent][type="primary"]):hover,
+        :host([transparent][type="primary"]):focus {
+            color: var(--color-pumpkin);
+        }
+        :host([transparent][type="secondary"]) {
+            color: var(--color-grassland);
+        }
+        :host([transparent][type="secondary"]):hover,
+        :host([transparent][type="secondary"]):focus {
+            color: var(--color-apple);
+        }
+        :host([transparent][type="tertiary"]) {
+            color: var(--color-grey);
+        }
+        :host([transparent][type="tertiary"]):hover,
+        :host([transparent][type="tertiary"]):focus {
+            color: var(--color-stone);
+        }
+        :host([transparent][type="warning"]) {
+            color: var(--color-cinnabar);
+        }
+        :host([transparent][type="warning"]):hover,
+        :host([transparent][type="warning"]):focus {
+            color: var(--color-flamingo);
+        }
+        :host([transparent][type="white"]) {
+            color: rgba(255, 255, 255, 1);
+        }
+        :host([transparent][type="white"]):hover,
+        :host([transparent][type="white"]):focus {
+            color: rgba(255, 255, 255, 0.8);
+        }
+        /** Disabled button */
+        :host([disabled]),
+        :host([disabled]):hover,
+        :host([disabled]):focus {
+            background-color: var(--color-grey);
+            cursor: default;
+        }
+        :host iron-icon {
+            display: inline-block;
+            height: 16px;
+            vertical-align: -5%;
+            width: 16px;
+        }
+    </style>
+    <iron-a11y-keys id="keys"
+                    keys="space enter"
+                    target="[[_target]]"
+                    on-keys-pressed="_keysPressed">
+                    </iron-a11y-keys>
+    <template is="dom-if" if="[[_displayIcon]]">
+        <iron-icon icon="kano-icons:[[icon]]"></iron-icon>
+    </template>
+    <slot id="content"></slot>
+  </template>
+
+  <script>
+    Polymer({
+      is: 'kwc-button',
+      properties: {
+            /**
+             * Boolean to indicate whether the button is `active` or `disabled`
+             */
+            disabled: {
+                type: Boolean,
+                value: false,
+                reflectToAttribute: true
+            },
+            _displayIcon: {
+                type: Boolean,
+                computed: '_iconProvided(icon)'
+            },
+            /**
+             * Boolean to indicate whether the button ghosted (semi-transparent)
+             */
+            ghost: {
+                type: Boolean,
+                value: false,
+                reflectToAttribute: true
+            },
+            /**
+             * String to indicate which button (if any) the button
+             * should display
+             */
+            icon: {
```

Comment:
Perhaps we should be more specific naming our apis. Since `icon` can be an id, the actual svg string, a path, an object, an object with all the previous options before, shall we call `iconId` instead? Just so we are 100% sure `iconId` is about the "icon id"?

Reply:
Yep, good point. I'll update that.